File: Atcoder_#353/javascript/b_atCoder_amusement_park.js
function main(input) {
  const lines = input.trim().split('\n');
  const [N, K] = lines[0].split(' ').map(Number);
  const A = lines[1].split(' ').map(Number);

  let startCount = 0;
  let seats = K;

  for (let i = 0; i < N; i++) {
    if (A[i] > seats) {
      startCount++;
      seats = K;
    }
    seats -= A[i];
  }

  if (seats < K) {
    startCount++;
  }

  console.log(startCount);
}

const fs = require('fs');
const input = fs.readFileSync('/dev/stdin', 'utf8');
main(input);

File: Atcoder_#353/ruby/b_atCoder_amusement_park.rb
n, k = gets.split.map(&:to_i)
a = gets.split.map(&:to_i)

start_count = 0
seats = k

a.each do |group_size|
  if group_size > seats
    start_count += 1
    seats = k
  end
  seats -= group_size
end

start_count += 1 if seats < k

puts start_count
